cms/navigation/wagtail_hooks.py

```diff
@@ -0,0 +1,32 @@
+from typing import TYPE_CHECKING, Optional
+
+from wagtail.permissions import ModelPermissionPolicy
+from wagtail.snippets.models import register_snippet
+from wagtail.snippets.views.snippets import SnippetViewSet
+
+from .models import MainMenu
+
+if TYPE_CHECKING:
+    from cms.users.models import User
+
+
+class NoAddModelPermissionPolicy(ModelPermissionPolicy):
+    """Model permission that doesn't allow creating more than one main menu instance."""
+
+    def user_has_permission(self, user: Optional["User"] = None, action: str | None = None) -> bool:
+        if action == "add" and MainMenu.objects.exists():
+            return False
+        return super().user_has_permission(user, action)
```

Comment:
```suggestion
        has_permission: bool = super().user_has_permission(user, action)
        return has_permission
```

should help with `Any from function declared to return "bool"  [no-any-return]`

Reply:
Done!